core/src/main/scala/kafka/server/DynamicConfigManager.scala

```diff
@@ -92,6 +93,7 @@ class DynamicConfigManager(private val zkUtils: ZkUtils,
           val entityType = map.get("entity_type") match {
             case Some(ConfigType.Topic) => ConfigType.Topic
             case Some(ConfigType.Client) => ConfigType.Client
+            case Some(ConfigType.Broker) => ConfigType.Broker
             case _ => throw new IllegalArgumentException("Config change notification must have 'entity_type' set to either 'client' or 'topic'." +
```

Comment:
This message doesn't mention `broker`. Another case where having a definition of supported types would make the code more robust.